### ISSUE TITLE
more reliably disable husky for ci/cd environments

### ISSUE THREAD

**Issue by dlants** [newrelic/node-native-metrics#195]
## Description

@newrelic/node-native-metrics is a dependency of one of my projects. When running `yarn install --verbose` on the project, building `node-native-metrics` failed with:

```
> yarn install --verbose

...

16:17:33 make: Leaving directory '/home/ubuntu/.cache/yarn/v6/.tmp/cded8684755c48d23b0d82fdddbfc97a.d0b66af020795b3a6552ea31dd3d07b1ba1471ef.prepare/node_modules/@newrelic/native-metrics/build'
16:17:33 gyp info ok 
16:17:33 install successful: _newrelic_native_metrics-7_1_2-native_metrics-83-linux-x64
16:17:33 success Saved lockfile.
16:17:33 $ husky install
16:17:33 husky - not a Git repository, skipping hooks installation
16:17:33 Build step 'Execute shell' marked build as failure
```

It seems like yarn tried to run the prepare script, which tried to run `husky install`. That in turn failed, because it did not find itself in a git repo.

This behavior on husky has been a pain point for many people, and they've tried to fix it a bunch of times, though it seems unreliably:

https://github.com/typicode/husky/issues/851

Their current docs provide several options for how to disable husky in various environments: https://typicode.github.io/husky/#/?id=disable-husky-in-cidockerprod

## Expected Behavior
We do not attempt to run husky when installing the package as a dependency.

## Your Environment
- yarn 1.22.18
- installing newrelic/native-metrics 7.1.2

**Comment by dlants** [comment]
I should mention, since this repo uses husky v7, I was able to work around this issue by setting:

```
export HUSKY_SKIP_INSTALL="true"
```

**Comment by bizob2828** [comment]
Thanks @dlants for capturing this issue.  It looks like you found a valid workaround for your issue.  I will talk to our PM to discuss getting this prioritized

**Comment by vidhya03** [comment]
We also faced similar issue. when we are using jhipster framework application. 


npm install -g
  ./gradlew -Pprod -Pwar bootWar --no-daemon -PbuildNumber=$GITHUB_RUN_NUMBER
  shell: /bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    JAVA_HOME_11_x64: /home/actions/actions-runner/_work/_tool/jdk/11.0.15/x64
    JAVA_HOME: /home/actions/actions-runner/_work/_tool/jdk/11.0.15/x64
    JAVA_HOME_11_X64: /home/actions/actions-runner/_work/_tool/jdk/11.0.15/x64
    NEXUS_USERNAME: ***
    NEXUS_PASSWORD: ***
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
npm ERR! code 1[27](https://github.softwareag.com/AIM/sci-mycloud/runs/916857?check_suite_focus=true#step:11:27)
npm ERR! path /home/actions/actions-runner/_work/sci-mycloud/sci-mycloud
npm ERR! command failed
npm ERR! command sh -c husky install
npm ERR! sh: husky: command not found

**Comment by jsumners-nr** [comment]
This should be resolved by #238
